tools.yml

```diff
@@ -0,0 +1,227 @@
+tools:
+  default:
+    cores: 1
+    mem: cores * 3.8
+    env: {}
+    params: {}
+    scheduling:
+      reject:
+        - offline
+    rules: []
+    rank: |
+      helpers.weighted_random_sampling(candidate_destinations)
+
+  # built in tools
+  Extract genomic DNA 1:
+    cores: 3
+  join1:
+    cores: 2
+    rules:
+    - if: input_size >= 10
+      cores: 5
+  tabular_to_csv:
+    rules:
+    - if: input_size >= 1
```

Comment:
We'll need to add unique ids to all rules, so that they can be overridden when inheriting.